ui.R

```diff
@@ -143,17 +156,105 @@ shinyUI(navbarPage(
+                                     
+                                   )
+                          ) ,
+                          
+                          tabPanel("Irrigation",
+                                   checkboxGroupInput("classType3", label = "Customer Class",inline=TRUE,
+                                                      choices = variables3, selected = "I1"),
+                                   fluidRow( column(12, plotlyOutput("revenue_time_series3", height=250) )),
+                                   fluidRow(
+                                     column(4,
+                                            plotlyOutput("barchart_by_tiers3", height=350),
+                                            radioButtons("barType", label = "",  selected = "Absolute", inline=TRUE,
+                                                         choices = list("Absolute" = "Absolute", "Percent" = "Percent"))
+                                     ),
+                                     
+                                     column(3,
+                                            plotlyOutput("fixed_revenue_barchart3", height=350)
+                                     ),
+                                     column(5, 
+                                            plotlyOutput("bill_change_boxplot3", height=100),
+                                            plotlyOutput("bill_change_histogram3", height=250) )
+                                     
+                                   )
+                          ), 
+                          tabPanel("Recycled",
+                                   checkboxGroupInput("classType4", label = "Customer Class",inline=TRUE,
+                                                      choices = variables4, selected = "RC1"),
+                                   fluidRow( column(12, plotlyOutput("revenue_time_series4", height=250) )),
+                                   fluidRow(
+                                     column(4,
+                                            plotlyOutput("barchart_by_tiers4", height=350),
+                                            radioButtons("barType", label = "",  selected = "Absolute", inline=TRUE,
+                                                         choices = list("Absolute" = "Absolute", "Percent" = "Percent"))
+                                     ),
+                                     
+                                     column(3,
+                                            plotlyOutput("fixed_revenue_barchart4", height=350)
+                                     ),
+                                     column(5, 
+                                            plotlyOutput("bill_change_boxplot4", height=100),
+                                            plotlyOutput("bill_change_histogram4", height=250) )
+                                     
+                                   )
+                          ),
+                          
```

Comment:
One thing to note is that your current approach of copying and pasting many times is probably not the best approach overall. This is the quickest and easiest way, and it is good for testing, but in the long run it has a few downsides.

1. It makes the code very bloated, so that there are many repetitive lines.
2. It requires a lot of copying and pasting that can be difficult to maintain and change. For example, if we wanted to go and add a new chart to one of the tabs, or to add a new tab, we would need to copy the old code, paste it, and a # to the end of the variable and hope we don't misstype anywhere.

Instead it would be great if the UI elements were generated in a loop(s)! There is a great example of how to do this at this link: http://shiny.rstudio.com/gallery/creating-a-ui-from-a-loop.html

You could even do the same thing with the dataframes like `df_plots` and `df_change`. This could be done by using a loop, and instead of storing 5 different dataframe ( `df_plots`,  `df_plots1`, etc...) you could use either a vector of dataframes, or a list of named dataframes. Something like...

```R
df_list <- List()
for(i in 1:5){
    df_list[[ paste("df_plots",i) ]] <- df %>% filter(...)  %>% ... 
}
```

This one would be a pretty big change, but it would keep the code much cleaner overall.